Assets/MixedRealityToolkit.SDK/Inspectors/UX/Interactable/InteractableInspector.cs

```diff
@@ -214,20 +214,91 @@ public virtual void RenderCustomInspector()
             }
             
             SerializedProperty dimensions = serializedObject.FindProperty("Dimensions");
+            // should be 1 or more
+            dimensions.intValue = Mathf.Clamp(dimensions.intValue, 1, 9);
+            int minMultiDimensionValue = 3;
+
+            // user-friendly dimension settings
+            string[] EnumNames = Enum.GetNames(typeof(Interactable.SelectionTypes));
+            int selectionTypeLength = EnumNames.Length;
+            int selectionTypeIndex = Mathf.Clamp(dimensions.intValue, 1, selectionTypeLength) - 1;
+
+            Interactable.SelectionTypes selectionType = Interactable.SelectionTypes.Button;
+            position = EditorGUILayout.GetControlRect();
+            GUIContent selectionTypeLabel = new GUIContent("Selection Mode", "How the Interactable should react to input");
             GUI.enabled = !isPlayMode;
-            EditorGUILayout.PropertyField(dimensions, new GUIContent("Dimensions", "Toggle or sequence button levels"));
+            EditorGUI.BeginProperty(position, selectionTypeLabel, dimensions);
+            {
+                selectionType = (Interactable.SelectionTypes)EditorGUI.EnumPopup(position, selectionTypeLabel, (Interactable.SelectionTypes)(selectionTypeIndex));
+                selectionTypeIndex = (int)selectionType;
+
+                if (selectionTypeIndex >= minMultiDimensionValue - 1)
+                {
+                    // multi dimension mode
+                    if (dimensions.intValue < minMultiDimensionValue)
```

Comment:
Why are you only updating dimensions.intValue if it's less than minMultiDimensionValue? Why not always update it?

Reply:
Great points, made updates as suggested.